lib/spring/generate/springInitializrGenerator.ts

```diff
@@ -92,66 +120,142 @@ function getUrlParameters(params: SpringInitializrProjectCreationParameters): an
-        defaultValue: "maven-project",
-    },
+    })
+    public projectType?: string;
 
-    dependencies: {
+    @Parameter({
         displayName: "Spring Boot dependencies",
         description: "comma separated list of dependencies",
         required: false,
-        defaultValue: "web,actuator",
-    },
+    })
+    public dependencies?: string;
+
+    @Parameter({
+        displayName: "Programming language",
+        description: "language (java, kotlin or groovy, default java)",
+        required: false,
+    })
+    public language?: string;
 
-    bootVersion: {
+    @Parameter({
         displayName: "Spring Boot version",
         description: "Version of Spring Boot you want to use",
         required: false,
-    },
+    })
+    public bootVersion?: string;
 
-    language: {
-        displayName: "Programming language",
-        description: "language (java, kotlin or groovy, default java)",
-        pattern: /^(java|kotlin|groovy)$/,
+    @Parameter({
+        displayName: "Packaging",
+        description: "packaging (jar or war, default jar)",
         required: false,
-        defaultValue: "java",
-    },
+    })
+    public packaging?: string;
 
-    javaVersion: {
+    @Parameter({
         displayName: "Java version level",
         description: "Java version level (8 or 10, default 8)",
-        pattern: /^(8|10)$/,
         required: false,
-        defaultValue: "8",
-    },
+    })
+    public javaVersion?: string;
 
-    packaging: {
-        displayName: "Packaging",
-        description: "packaging (jar or war, default jar)",
-        pattern: /^(jar|war)$/,
+    @Parameter({
+        ...MavenGroupIdRegExp,
+        required: true,
+        order: 50,
+    })
+    public groupId: string;
+
+    @Parameter({
+        ...JavaPackageRegExp,
+        required: true,
+        order: 53,
+    })
+    public rootPackage: string;
+
+    @Parameter({
+        pattern: /.*/,
+        description: "Version to use",
+        required: false,
+    })
+    public version: string = "0.1.0-SNAPSHOT";
+
+    @Parameter({
+        displayName: "Class Name",
+        description: "name for the service class",
+        ...JavaIdentifierRegExp,
+        required: false,
+    })
+    public enteredServiceClassName?: string;
+
+    public addAtomistWebhook: boolean = true;
+    public source: RemoteLocator;
+    public target: RepoCreationParameters;
+    public description?: string;
+
+    @Parameter({
+        ...MavenArtifactIdRegExp,
+        displayName: "artifactId",
         required: false,
-        defaultValue: "jar",
-    },
+        order: 51,
+    })
+    public enteredArtifactId?: string = "";
 
-    ...SpringProjectCreationParameterDefinitions,
-};
+    public bindAndValidate(): ValidationResult | Promise<ValidationResult> {
+        const validationErrors = [];
+        if (this.bootVersion) {
+            const springBootVersions = metaData.bootVersion.values.map((v: any) => v.id) as string[];
```

Comment:
Are all these metaData properties guaranteed to be defined?

Reply:
yes, according to the Spring Initializr documentation, as long as the mime type is provided